enhancements/cvo/cluster-profiles.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Cluster Profiles
+authors:
+  - "@csrwng"
+reviewers:
+  - "@abhinavdahiya"
+  - "@crawford"
+  - "@sdodson"
+  - "@derekwaynecarr"
+  - "@smarterclayton"
+approvers:
+  - "@derekwaynecarr"
+  - "@smarterclayton"
+creation-date: 2020-02-04
+last-updated: 2019-02-04
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# cluster-profiles
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+Cluster profiles are a way to support different deployment models for OpenShift clusters. 
+A profile is an identifier that the Cluster Version Operator uses to determine
+which manifests to apply. Operators can be excluded completely or can have different
+manifests for each supported profile.
+
+## Motivation
+
+In order to support different deployment models in which not all operators rendered by
```

Comment:
is there any other known use case for profiles we can include here?
What do we consider best practices for profiles vs no-op operators? e.g do we want a "baremetal profile" or do we want no-op "baremetal" operator deployed everywhere?

Reply:
I agree it would be good to include other scenarios we want this design to cover and the bare metal one is a good one consider

Perhaps "no-op operator" isn't a good way to classify the baremetal-operator case though - for example, do we want bare metal specific CRDs to be installed on all platforms? It's not just a question of a process running that does nothing ...

Reply:
@markmc i think its an error that we have namespaces and crds deployed to a cluster for contexts that are not appropriate.  we should aspire to move away from that rather than continue to lean into it.  for example, every cluster has a `openshift-kni-infra` or `openshift-ovirt-infra` even where it is not appropriate.

the initial use case for cluster profile was to exclude specific operators entirely from being applied.  the most notable example was the cluster version operator not installing kube-* operators in deployments where the control plane is hosted on an external supervisor cluster a la IBM Public Cloud.

  for example, we have a kni namespace or an openstack namespace on every openshift cluster.

Reply:
(feedback taken - #212 won't pursue the cluster profile path, instead it will propose a new SLO for bare metal)

But on the point of CRDs ... your POV seems at odds with Clayton's example of the CNO in this comment: https://github.com/openshift/enhancements/pull/212#issuecomment-587495821

Would be great to get a resolution on this, because the path of a bare metal SLO that is installed, running, but "disabled" on other platforms would seem to inevitably be introducing more stuff "deployed to a cluster for contexts that are not appropriate".